src/dodal/devices/i22/dcm.py

```diff
@@ -0,0 +1,152 @@
+import time
+from dataclasses import dataclass
+from typing import Dict, Literal
+
+from bluesky.protocols import Reading
+from event_model.documents.event_descriptor import DataKey
+from ophyd_async.core import ConfigSignal, StandardReadable, soft_signal_r_and_setter
+from ophyd_async.epics.motion import Motor
+from ophyd_async.epics.signal import epics_signal_r
+
+# Conversion constant for energy and wavelength, taken from the X-Ray data booklet
+# Converts between energy in KeV and wavelength in angstrom
+_CONVERSION_CONSTANT = 12.3984
+
+
+@dataclass(frozen=True, unsafe_hash=True)
+class CrystalMetadata:
+    """
+    Metadata used in the NeXus format,
+    see https://manual.nexusformat.org/classes/base_classes/NXcrystal.html
+    """
+
+    usage: Literal["Bragg", "Laue"] | None = None
+    type: str | None = None
+    reflection: tuple[int, int, int] | None = None
+    d_spacing: float | None = None
+
+
+class DoubleCrystalMonochromator(StandardReadable):
```

Comment:
How i22 specific is this implementation?
Currently crystal_1_metadata, crystal_2_metadata are optional: as this is in the i22 subdirectory I think it's reasonable to make them required args, or else move out of this subdirectory (assuming the signal names are consistent elsewhere)

Reply:
They are optional because they default to `None`, i.e. nothing breaks if the information isn't there for whatever reason. I think that's correct because it's not mission-critical